scripting/FileParsing.js

```diff
@@ -1 +1,141 @@
+      //there are commas within cells so it switches them with "|" to avoid splitting the cells
+      let isComma = false;
+      for(y = 0; y < array.length; y++){
+          for(x = 0; x < array[y].length; x++){
+              if(array[y].charAt(x) == "\""  && !isComma){
+                  isComma = true;
+              }
+              else if(array[y].charAt(x) == "\""  && isComma){
+                  isComma = false;
+              }
+              if(isComma && array[y].charAt(x) == ","){
+                  array[y] = array[y].replaceAt(x,"^");
+              }
+          }
+      }
+      //breaks array into multidimentional array - essentially a grid same as the excel doc
+      for(let x = 0; x < array.length; x++){
+          array[x] = array[x].split(",");
+      }
+      //finds which column has mod_id and deletes all the garbage before it, deletes everything between mod_id and AM, deletes everything after the last loci
+      let correctColumn = 0;
+      while(array[0][correctColumn] != "mod_id"){
+          correctColumn ++;
+      }
+      for(let y = 0; y < array.length; y++){
+          array[y].splice(0,correctColumn);
+      }
+      correctColumn = 0;
+      while(array[0][correctColumn] != "AM"){
+          correctColumn ++;
+      }
+
+      for(let y = 0; y < array.length; y++){
+          array[y].splice(1,correctColumn - 1);
+      }
+      correctColumn = 0;
+      for(x = 0; x < array[0].length; x++){
+          if(array[0][x].slice(0,3) == "mod" && array[0][x] != "mod_id"){
+              correctColumn = x;
+          }
+      }
+      if(correctColumn > 0){
+          for(let y = 0; y < array.length; y++){
+              array[y].splice(correctColumn,array[0].length - 1);
+          }
+      }
+
+      for(y = 0; y < array.length; y++){
+          for(x = 0; x < array[y].length; x++){
+              for(z = 0; z < array[y][x].length; z++){
+                  if(array[y][x].charAt(z) == "\""){
+                      array[y][x] = array[y][x].substr(1);
+                      array[y][x] = array[y][x].substr(0, array[y][x].length - 1);
+                  }
+              }
+          }
+      }
+      for(y = 0; y < array.length; y++){
+          for(x = 1; x < array[y].length; x++){
+              array[y][x] = array[y][x].split("^");
+          }
+      }
+      //object maker
+      let objArray = [];
+      let loci = {};
+      for(let y = 1; y < array.length; y++){
+          tempObj = {}
+          loci  = {};
+          for(x = 1; x < array[y].length; x++){
+              loci[array[0][x]] = [array[y][x]];
+          }
+          tempObj.loci = loci;
+          tempObj.modelIdentification = array[y][0];
+          objArray[y - 1] = tempObj;
+      }
+      for(let y = 0; y < objArray.length; y++){
+          if(objArray[y].modelIdentification === "" || objArray[y].modelIdentification === undefined || objArray[y].modelIdentification === null || objArray[y].modelIdentification === "<empty string>"){
```

Comment:
Very long logic statement, maybe should be written in multiple lines or make a function for it.